app/models/registration.rb

```diff
@@ -409,42 +403,100 @@ def consider_auto_close
-             .joins(:competition)
-             .order(:start_date)
-    end
+    # There are a limited number of "reserved" spots for newcomer_month_eligible competitions
+    # We know that there are _some_ available_spots in the comp available, because we passed the competitor_limit check above
+    # However, we still don't know how many of the reserved spots have been taken up by newcomers, versus how many "general" spots are left
+    # For a non-newcomer to be accepted, there need to be more spots available than spots still reserved for newcomers
+    return if available_spots > competition.newcomer_month_reserved_spots_remaining
 
-    # this relies on the scopes being named the same as `checked_status` but it is a significant performance improvement
-    sibling_registrations.send(registration_status)
+    errors.add(:competing_status, :exceeding_newcomer_limit, frontend_code: Registrations::ErrorCodes::NO_UNRESERVED_SPOTS_REMAINING)
+  end
+
+  delegate :competitor_limit_enabled?, :enforce_newcomer_month_reservations?, to: :competition
+
+  validate :cannot_exceed_competitor_limit, if: [:trying_to_accept?, :competitor_limit_enabled?]
+  private def cannot_exceed_competitor_limit
+    return unless competition.registrations.accepted_and_competing_count >= competition.competitor_limit
+
+    errors.add(
+      :competing_status,
+      :exceeding_competitor_limit,
+      frontend_code: Registrations::ErrorCodes::COMPETITOR_LIMIT_REACHED,
+    )
+  end
+
+  validate :only_one_accepted_per_series, if: [:part_of_competition_series?, :trying_to_accept?]
+  private def only_one_accepted_per_series
+    return unless series_sibling_registrations.accepted.any?
+
+    errors.add(
+      :competition_id,
+      :already_registered_in_series,
+      message: I18n.t('registrations.errors.series_more_than_one_accepted'),
+      frontend_code: Registrations::ErrorCodes::ALREADY_REGISTERED_IN_SERIES,
+    )
   end
 
-  SERIES_SIBLING_DISPLAY_STATUSES = [:accepted, :pending].freeze
+  delegate :part_of_competition_series?, to: :competition, allow_nil: true
 
-  def series_registration_info
-    SERIES_SIBLING_DISPLAY_STATUSES.map { |st| series_sibling_registrations(st) }
-                                   .map(&:count)
-                                   .join(" + ")
+  def series_sibling_registrations
+    return [] unless part_of_competition_series?
+
+    competition.series_sibling_registrations
+               .where(user_id: self.user_id)
   end
 
   def ensure_waitlist_eligibility!
     raise ArgumentError.new("Registration must have a competing_status of 'waiting_list' to be added to the waiting list") unless
       competing_status == Registrations::Helper::STATUS_WAITING_LIST
   end
 
+  def trying_to_cancel?
+    competing_status_changed? && (competing_status_cancelled? || competing_status_rejected?)
+  end
+
+  validate :not_changing_events_when_cancelling, if: [:trying_to_cancel?, :tracked_event_ids?, :competition_events_changed?]
```

Comment:
Do we need `:tracked_event_ids?` in here? It seems to be checking for presence, but 
- the way tracked_event_ids is defined makes me think it will always be present? There's no way for a registration to be created without event_ids? 
- Shouldn't `competition_events_changed?` return `false` if there are no `tracked_event_ids`?

Reply:
Yes, we do need it, because `competition_events_changed?` otherwise triggers too early. It has a side-effect due to the inherently caching nature of Rails, and the only way to circumvent that would be by reloading the association fresh every time the comparison is run. In the big picture, that would be sinfully expensive in terms of DB calls.

Briefly put, think of this presence check as "Do we have anything to compare against in the first place?". And only if the answer to that question is Yes, we proceed with running the actual comparison.

(The last elements of the `if` array don't get checked as soon as one of them is falsey)

Reply:
How would we end up in the situation where we don't have anything to compare against - ie, where `tracked_event_ids` is `nil`?

Reply:
Virtually anywhere in our code. The only case when we do have something to compare is when we intentionally write the value from the outside (`attr_writer` above). And currently, that only happens in `registration_checker`s request loading routine. All other APIs that do somethingTM with registrations don't write this and thus shouldn't be compared.

For reference, even a simple `FactoryBot.create(:registration)` in our test suite triggers validations, and it will needlessly run this comparison if the check you're commenting on isn't there.

Reply:
This took me a second to understand, so posting my explanation here in case anyone looks at this thread in the future: 
- We have the function tracked_event_ids (line 484)
- Separately, we have the instance variable @tracked_event_ids - this is what the attr_writer accesses
- tracked_event_ids method will return @tracked_event_ids if it is present, otherwise it will fall back to the registrations event_ids association
- tracked_event_ids? is checking the present of the @tracked_event_ids instance variable specifically - and in cases where that variable is nil, no further comparison/action is needed